src/app/App.tsx

```diff
@@ -1,30 +1,42 @@
 import { css } from '@emotion/css';
 import { GrafanaTheme2, PageLayoutType } from '@grafana/data';
 import { PluginPage } from '@grafana/runtime';
-import { useStyles2 } from '@grafana/ui';
+import { ErrorBoundary, useStyles2 } from '@grafana/ui';
 import { queryClient } from '@shared/infrastructure/react-query/queryClient';
+import { initFaro } from '@shared/infrastructure/tracking/faro/faro';
 import { QueryClientProvider } from '@tanstack/react-query';
-import React from 'react';
+import React, { useState } from 'react';
 
 import { Onboarding } from './components/Onboarding/Onboarding';
 import { Routes } from './components/Routes/Routes';
-import './infrastructure/faro';
+import { ErrorPage } from './ui/ErrorPage';
+
+initFaro();
 
 export function App() {
   const styles = useStyles2(getStyles);
+  const [error, setError] = useState<Error>();
+
+  if (error) {
+    return <ErrorPage error={error} />;
+  }
 
   return (
-    <QueryClientProvider client={queryClient}>
-      <Onboarding>
-        <div className={styles.pageContainer}>
-          <PluginPage layout={PageLayoutType.Custom}>
-            <div className="pyroscope-app">
-              <Routes />
+    <ErrorBoundary onError={setError}>
```

Comment:
`ErrorBoundary` from `@grafana/ui` pushes to faro already in  https://github.com/grafana/grafana/blob/47f87171496f675bc44ddf839effaed1846bfa4e/packages/grafana-ui/src/components/ErrorBoundary/ErrorBoundary.tsx#L38. But how does it work when faro is initialized twice - in grafana and in the app? Is the same error reported twice?